ENIGMAsystem/SHELL/Audio_Systems/DirectSound/DSsystem.cpp

```diff
@@ -70,7 +70,7 @@ void eos_callback(void* soundID, unsigned src) {
 }
 
 int audiosystem_initialize() {
-  if (get_window_handle() == NULL) {
+  if (enigma::hWnd == NULL) {
```

Comment:
Otherwise, in addition to the breakages, we end up with this line. This is the kind of inter-system dependency I was talking about, earlier. Using `window_handle()` was *somewhat* questionable, but part of ENIGMA's API. Using `enigma::hWnd` is not portable. For example, this system *would* have worked with SDL before. Now it can't, unless SDL *happens* to store a variable by the same name. If that name is changed, this relationship will be broken. Note that SDL is an example. I am not asking you to make it so that using enigma::hWnd works here when building against SDL, but instead to use library methods so arbitrary components will play nicely together.

Reply:
SDL does story a variable of the same name because in this pr I added it. There's nothing breaking here. I've already tested it on Linux. If you feel more comfortable with it I can test on Windows too before we hit merge but not only did the appveyor CI pass, I have looked over this code a crap ton of times now to make sure everything is correct based.